elastic/kibana
# [Fleet] Use dns.lookup to determine whether we should fetch from product version API

## Summary

Fixes https://github.com/elastic/kibana/issues/173125 (hopefully)

Add a DNS lookup before fetching from the `product_versions` API endpoint to fail fast in airgapped environments where blocked requests are allowed to time out. 

I'm still experimenting with ways to simulate an environment in which blocked requests will time out, instead of being rejected instantly at the firewall level. If anyone has any suggestions for that I'd appreciate it. 

## Testing instructions

1. Disable internet entirely on your workstation
2. Click the "add agent" flyout in Fleet
3. Set `ignoreCache = true` in the `services/agent/versions.ts` file to prevent caching responses and to always attempt a refetch
4. Observe the debug log that the API request for available versions was skipped because the DNS lookup failed
5. Observe that the enrollment flyout still renders as expected with the latest available version

### Checklist

Delete any items that are not applicable to this PR.
